crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -203,13 +204,22 @@ pub(crate) fn infer_expression_types<'db>(
 /// This is a small helper around [`infer_expression_types()`] to reduce the boilerplate.
 /// Use [`infer_expression_type()`] if it isn't guaranteed that `expression` is in the same file to
 /// avoid cross-file query dependencies.
+///
+/// `query_file` is the file for which the current query performs type inference.
+/// It acts as a token of prove that we aren't accessing an AST node from a different file
+/// than in which the current enclosing Salsa query (which would lead to cross-file dependencies).
```

Comment:
```suggestion
/// `query_file` is the file for which the current query performs type inference.
/// It acts as a token of proof that we aren't accessing an AST node from a different file
/// to the one from which the current enclosing Salsa query is being called.
/// Doing so would lead to cross-file dependencies, hurting incremental computation.
```